docs/architecture/adr-009-non-interactive-default-rules-for-reduced-padding.md

```diff
@@ -0,0 +1,177 @@
+- How big is the proof size for this message?
+- What is the worst constructible block with the most amount of padding with old and new non-interactive defaults?
+- What is the quantified padding and proof size cost?
+
+## 1. Given square size independent commitments, why does a message in a larger square size result in an O(log(n)) message inclusion proof size?
+
+If you use the current non-interactive default rules then the message begins at a location aligned with the largest power of 2 that is not larger than the message length or k. Because the subtree roots are aligned you can skip some subtree roots and calculate their parents.
+In the example below instead of proving H1, H2, H3, and H4 to the DataRoot you can prove H10. **H10 is part of the commitment generation and part of the Merkle tree to the DataRoot.** That is why you can use it for more efficient proofs. In smaller square sizes, you cannot do this, because H10 does not exist. The nodes in **red** are the subtree nodes that you need to provide for the message inclusion proof. The nodes in **blue** are the additional nodes for the Merkle proof.
+
+![Efficient Merkle Proofs with ADR008](./assets/effizicient-proof-size-ADR008.png)
+
+So why can you not do it with the proposed non-interactive default rules? This is because H10 is not generated. In the diagram below the first 8 shares are in the row before and therefore the tree structure changes. The previous subtree root H10 is now H23 and cannot be used for the efficiency boost.
+The commitment is still the same but we need to use the bottom subtree roots for the message inclusion proof.
+
+![Shifted Message](./assets/new-ni-rules-message-shift.png)
+
+## 2. Assuming message inclusion proof sizes change from O(log(n)) to O(sqrt(n)), what is the worst-case constructible message?
+
+Given a square size k, the biggest message that you can construct that is affected by the proposed non-interactive default rules has a size (k/2)². If you construct a message that is bigger than (k/2)² the `minSquareSize` will be k. If the minSquareSize is k in a square of size k then the current non-interactive default rules are equivalent to the proposed non-interactive default rules, because the message starts always at the beginning of a row. In other words, if you have k² shares in a message the worst constructible message is a quarter of that k²/4, because that is the size of the next smaller square.
+
+If you choose k²/4 as the worst constructible message it would still have O(sqrt(n)) subtree roots. This is because the size of the message is k²/4 with a width of k and a length of k/4. This means the number of rows the message fills approaches O(sqrt(n)). Therefore we need to find a message where the number of rows is log(n) of the size of the message.
+
+With k being the square size and n being the number of shares and r being the number of rows, we want to find a message so that:
+k * r = n & log(n) = r => k = n/log(n)
```

Comment:
So we want to find a message of size n where the rows of the message are log(n) so we still have O(log(n)) proof size. This means that r = log(n) . k is the square size , so a rollup message is roughly k (width) *r(height) = n (Area). I plugged the 2 equations together and i got a relationship between square size and message size to resolve this. As k is fixed 2, 4, 8 etc we can calculate n and in turn with n we can calculate r for k = 2,4,8 etc. 
I must say though that we could have had for example any c for r = c * log(n) as it would still be O(log(n)). But as we don't increase blocksize to infinity, we need to find a compromise between O-Notation and quantitative solution and mixing them here might create some confusion. Maybe r = k/4 is a better candidate for analysis and that is still up for debate. Happy for input @musalbas